### PR TITLE
Add echoshader refactor blog

### PR DESCRIPTION
**Pull request by ldr426**

### CODE REVIEW

File: content/post/2023-09-18-echoshader-refactor-intro/index.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+---
+# Documentation: https://sourcethemes.com/academic/docs/managing-content/
+title: "Hello from Dingrui Lei, A Contributor Refactoring Work of Echoshader!"
+date: 2023-09-18T00:00:00-08:00
+authors:
+  - Dingrui Lei
+slug:
+draft: false
+tags:
+summary: "Echospace hosted a contributor to jump refactor [echoshader](https://github.com/OSOceanAcoustics/echoshader), a package for interactive visualization of echosounder data."
```
Comment (valentina-s) [inline content/post/2023-09-18-echoshader-refactor-intro/index.md:10]: "jump refactor" may not be clear. You could use refactor, or revamp, upgrade?

File: content/post/2023-09-18-echoshader-refactor-intro/index.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+_[Dingrui Lei](https://github.com/ldr426) is our contributor._
+
+-------------------------------
+# My 2023 Summer Internship with Echoshader: A Dive into Advanced Ocean Sonar Data Visualization
+Author: [Dingrui Lei](mailto:leidingrui426@gmail.com)
+[Slides](https://docs.google.com/presentation/d/1HmL2-luVmA9T5HfS3L1kBu8c7dDHo75znwaS-8YlTSE/edit#slide=id.p) of presentation 
+[Docs](https://echoshader--140.org.readthedocs.build/en/140/intro.html)  for this version
+
+Hello, readers! I'm excited to share my summer internship experience working on the fascinating project, Echoshader. This Python package, designed to enhance the visualization of ocean sonar data, has been my focus this summer. While I won't be delving into technical jargon, I'll give you a glimpse of my journey, the challenges I faced, and the accomplishments achieved during my internship.
```
Comment (valentina-s) [inline content/post/2023-09-18-echoshader-refactor-intro/index.md:41]: You could add a line here that the prototype was built during GSoC'22.

File: content/post/2023-09-18-echoshader-refactor-intro/index.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+I defined the class and set up initial values and interactive widgets. These widgets allow users to tweak parameters and explore data interactively.
+
+### 2. Callbacks and Streams: Making It Interactive
+
+Echoshader needed to be interactive, allowing users to explore data dynamically. This required creating callback methods and stream objects. These elements connected user interactions to visualization updates, making the whole experience smooth and intuitive.
+
+### 3. Extending Xarray with Accessors: A New Level of Functionality
+
+One of the exciting challenges I encountered was extending xarray's functionality using accessors. This means adding custom methods and functionality to xarray objects, without cluttering the code with custom functions. We created a custom "eshader" accessor, which allowed us to take echogram visualization to the next level.
```
Comment (valentina-s) [inline content/post/2023-09-18-echoshader-refactor-intro/index.md:65]: you can add quotes to `xarray` so it is clear it is a package.

File: content/post/2023-09-18-echoshader-refactor-intro/index.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+-   Fisheries scientists rely on echograms to identify fish aggregations, scrolling through data collected on ships to assess populations.
+-   Oceanographers use echograms to observe zooplankton movements in mooring data over extended periods.
+-   The "tricolor" echogram helps distinguish different fish species, thanks to its clever mapping of three frequencies to RGB colors.
+
+## Tracking and Curtain Visualization
+
+One of the most exciting aspects of Echoshader is tracking and curtain visualization. It's like having a GPS for underwater data.
+
+-   Echogram-Control Mode: Visualizing data on a map helps assess fish associations with environmental variables.
```
Comment (valentina-s) [inline content/post/2023-09-18-echoshader-refactor-intro/index.md:79]: How about makeing the titles of the bullet points in bold. Then they will be more visible.

File: content/post/2023-09-18-echoshader-refactor-intro/index.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+-   Track-Control Mode: Highlighting ship track sections on the map while viewing corresponding echograms offers precise insights into marine life at specific locations.
+-   Curtain Visualization: Representing longer data sections as curtains provides a broader spatial perspective on fish aggregations.
+
+## Histograms and Statistics Tables: Tools for Deeper Analysis
+
+Histograms and statistics tables are essential for fisheries scientists.
+
+-   Focused Analysis: Scientists can zoom in on specific data sections to examine volume backscattering strength (Sv) distribution and understand the types of fish present.
```
Comment (valentina-s) [inline content/post/2023-09-18-echoshader-refactor-intro/index.md:87]: Those titles can be bold too.

**Review by valentina-s — COMMENTED** [review]
Thanks for writing the blog @ldr426, looks good! I have a few small suggestions inline. Consider also adding at least one picture (maybe a collage of the plots? or maybe one of the numerous whiteboard snapshots? whatever you like).

**Comment by ldr426** [comment]
> Thanks for writing the blog @ldr426, looks good! I have a few small suggestions inline. Consider also adding at least one picture (maybe a collage of the plots? or maybe one of the numerous whiteboard snapshots? whatever you like).

Thanks @valentina-s, I have improved the article based on your feedback. Could you check why it took long time to build testing and then the testing just was cancels the testing without any errors or warnings shown? I have checked my article and found no syntax or format error.

**Review by valentina-s — APPROVED** [review]

**Pull request merged**